tycho-artifactcomparator/src/main/java/org/eclipse/tycho/zipcomparator/internal/TextContentsComparator.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 Hannes Wellmann and others.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Hannes Wellmann - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.tycho.zipcomparator.internal;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import org.codehaus.plexus.component.annotations.Component;
+import org.eclipse.tycho.artifactcomparator.ArtifactComparator.ComparisonData;
+import org.eclipse.tycho.artifactcomparator.ArtifactDelta;
+import org.eclipse.tycho.artifactcomparator.ComparatorInputStream;
+
+@Component(role = ContentsComparator.class, hint = TextContentsComparator.TYPE)
+public class TextContentsComparator implements ContentsComparator {
```

Comment:
You might want to move the code from org.eclipse.tycho.zipcomparator.internal.XmlComparator.createDelta(String, ComparatorInputStream, ComparatorInputStream) into this calss as a static method that can be used to generate unified diff for different text formats